src/tera.rs

```diff
@@ -306,11 +309,31 @@ impl Tera {
         renderer.render()
     }
 
+    /// Renders a one off template (for example a template coming from a user
+    /// input) given a `Context` and an instance of Tera. This allows you to
+    /// render templates using custom filters or functions.
+    ///
+    /// Any errors will mention the `__tera_one_off` template: this is the name
+    /// given to the template by Tera.
+    ///
+    /// ```rust,ignore
+    /// let mut context = Context::new();
+    /// context.insert("greeting", &"Hello");
+    /// let string = tera.render_str("{{ greeting }} World!", &context)?;
+    /// assert_eq!(string, "Hello World!");
+    /// ```
+    pub fn render_str(&mut self, input: &str, context: &Context) -> Result<String> {
+        self.add_raw_template(ONE_OFF_TEMPLATE_NAME, input)?;
+        let string = self.render(ONE_OFF_TEMPLATE_NAME, &context)?;
```

Comment:
the template will still be there if the render fails, it needs to remove the template when it errors as well.

Reply:
Updated so now that it is always removed.